From 1eefdeb87cfe46f790bc68346bc858f3bdcce9ff Mon Sep 17 00:00:00 2001
From: Lai Jun Ann <jun.ann.lai@intel.com>
Date: Thu, 28 Jul 2022 15:09:56 +0800
Subject: [PATCH] tc: fix no error return with large parent id used

To fix the issue that large value for parent id is not handling properly
by modifying the datatype of maj and min to unsigned long as the datatype
that returned by strtoul() is unsigned long.

Signed-off-by: Zulkifli, Muhammad Husaini <muhammad.husaini.zulkifli@intel.com>
Signed-off-by: Lai Jun Ann <jun.ann.lai@intel.com>
---
 tc/tc_util.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tc/tc_util.c b/tc/tc_util.c
index ba34aed..14021df 100644
--- a/tc/tc_util.c
+++ b/tc/tc_util.c
@@ -93,7 +93,7 @@ ok:
 
 int get_tc_classid(__u32 *h, const char *str)
 {
-	__u32 maj, min;
+	unsigned long maj, min;
 	char *p;
 
 	maj = TC_H_ROOT;
-- 
2.17.1

